### PR TITLE
U674/t689 compile shader once

### PR DESCRIPTION
**Pull request by Praccen**

### CODE REVIEW

**Comment by Praccen** [comment]
Needs fixing. Don't accept.

File: SPLASH/res/shaders/compute/ParticleComputeShader.hlsl [hunk 1]
```diff
@@ -117,12 +118,12 @@ void updateAnimation(int particleIndex) {
...
-    int ai = (CSPhysicsBuffer[particleIndex].animationIndexCounter += inputBuffer.frameTime*24.f) % (4 * textureSize);
+    int atlasSizeX = inputBuffer.atlasSize.x;
+    int ai = (CSPhysicsBuffer[particleIndex].animationIndexCounter += inputBuffer.frameTime*24.f) % (inputBuffer.atlasSize.y * atlasSizeX);
 
-    float offset = 1.0f / textureSize;
-    float ix = ai % textureSize;
-    float iy = (int)(ai / textureSize);
+    float offset = 1.0f / atlasSizeX;
+    float ix = ai % atlasSizeX;
+    float iy = (int)(ai / atlasSizeX);
```
Comment (whalemane) [inline SPLASH/res/shaders/compute/ParticleComputeShader.hlsl]: ai and atlasSizeX are both int, so this is probably nothing important, but feels a bit wonky with all the casting and division between types

File: Sail/src/Sail/entities/EntityFactory.cpp [hunk 2]
```diff
@@ -48,9 +48,10 @@ void EntityFactory::CreateCandle(Entity::SPtr& candle, const glm::vec3& lightPos
 	particleEmitterComp->constantVelocity = { 0.0f, 0.2f, 0.0f };
 	particleEmitterComp->acceleration = { 0.0f, 1.0f, 0.0f };
 	particleEmitterComp->spread = { 0.1f, 0.1f, 0.1f };
-	particleEmitterComp->spawnRate = 0.001f;
+	particleEmitterComp->spawnRate = 0.01f;
```
Comment (whalemane) [inline Sail/src/Sail/entities/EntityFactory.cpp]: Is this right or something used for testing?

File: Sail/src/Sail/entities/components/ParticleEmitterComponent.cpp [hunk 3]
```diff
@@ -216,7 +230,8 @@ void ParticleEmitterComponent::updateOnGPU(ID3D12GraphicsCommandList4* cmdList,
 		// Update nr of particles in this buffer and erase added and removed emitters from the queues
 		m_cpuOutput[context->getSwapIndex()].previousNrOfParticles = glm::min(m_cpuOutput[context->getSwapIndex()].previousNrOfParticles + numPart - numPartRem, data.outputVertexBufferSize / 6);
 
-		m_cpuOutput[context->getSwapIndex()].newParticles.erase(m_cpuOutput[context->getSwapIndex()].newParticles.begin(), m_cpuOutput[context->getSwapIndex()].newParticles.begin() + numPart);
+		//m_cpuOutput[context->getSwapIndex()].newParticles.erase(m_cpuOutput[context->getSwapIndex()].newParticles.begin(), m_cpuOutput[context->getSwapIndex()].newParticles.begin() + numPart);
```
Comment (whalemane) [inline Sail/src/Sail/entities/components/ParticleEmitterComponent.cpp]: Remove this?

**Pull request merged**